docs/DeveloperGuide.md

```diff
@@ -1,29 +1,115 @@
+It then updates the ```quantity``` variable according to the quantity inputted by the user.
+
+
+#### Preventing duplicate items
+
+When using the `add` function, we have decided to implement a passive function that checks whether the item with the same name already exists in the packingList.
+
+This is done through the `contains()` method in class `PackingList()`, which is called during `execute` in an `AddCommand` object.
+
+Below is the UML diagram showing what occurs during `add` function.
+![img_1.png](img_1.png)
+When `execute()` is called in `a`(object of class `AddCommand`), the `addItem(item)` method is called in the object `packingList`. This method will see if method `contains()` will return `true`.
+
+The `contains()` method, which is shown as a reference on the right of the UML diagram, is a boolean method that loops through all items in the packingList, and returns `true` if any of the existing items have the same description as the item to be added in `toAdd`, and `false` otherwise.
+
+When `contains()` returns `true`, method `addItem(item)` will be interrupted and a `DuplicateItemException` will be thrown from `packingList`, which will be caught by `a`. If `contains()` returns false, the item will be added onto `packingList`.
+
+In both scenarios, `ui.printToUser` will be called to print a message to the user. In the former case, `ExistItemError` will be printed, while `AddSuccess` will be shown if the item was added with no issues.
+
+#### Delete Command
+
+Delete command is used to delete an item from the packing list.
+
+Mechanism: ```DeleteCommand.execute()``` calls the ```PackingList.deleteItem()``` method from the ```PackingList``` class which executes the ```ArrayList.remove()``` method to remove the item from the ```PackingList``` ArrayList.
+
+
+#### Help Command
+Help command is used to exit the BagPacker application.
+
+Execute: ```HelpCommand.execute()``` prints the following help message.
+
+```
+All Commands:
+1. add : Adds a quantity of items to the packing list.
+	Example: add 3 toothbrushes
+2. delete : Deletes an item from the packing list.
+	Example: delete 1
+3. list : List all items in packing list.
+	Example: list
+4. pack : Marks an item as packed in the packing list.
+	Example: pack 2 of 3
+	Meaning: packs 2 quantities of the third item in the packing list
+5. unpack : Marks an item as unpacked in the packing list.
+	Example: unpack 1 of 2
+	Meaning: unpacks 1 quantity of the second item in the packing list
+6. bye : Stops the BagPacker Application
+	Example: bye
+____________________________________________________________
+
+```
+
+#### Bye Command
+```ByeCommand``` is used to exit the BagPacker application.
+
+Mechanism: ```ByeCommand.execute()``` updates the static boolean ```isBagPackerRunning``` to be false. 
```

Comment:
In which class is the `isBagPackerRunning` static variable stored? Would it be a good idea to specify?